webcam-capture-drivers/driver-raspistill/src/main/java/com/github/sarxos/webcam/ds/raspistill/RaspistillDevice.java

```diff
@@ -0,0 +1,252 @@
+		}
+		isOpen=false;
+	}
+	
+	@Override
+	public void dispose() {
+		arguments=null;
+		service=null;
+		process=null;
+	}
+	//TODO
+	@Override
+	public BufferedImage getImage() {
+		return null;
+	}
+
+	@Override
+	public String getName() {
+		if(LOGGER.isDebugEnabled()) {
+			LOGGER.debug("Now only one simple name retrieved, it is not real camera hardware vendor's name.");
+		}
+		return "raspistill Camera App@"+this.cameraSelect;
+	}
+	
+	@Override
+	public Dimension getResolution() {
+		if(dimension==null) {
+			dimension=new Dimension(Integer.parseInt(arguments.get("width")), Integer.parseInt(arguments.get("height")));
+		}
+		return dimension;
+	}
+	
+	@Override
+	public Dimension[] getResolutions() {
+		if(LOGGER.isDebugEnabled()) {
+			LOGGER.debug("now return one dummy dimension array of current setting, heardware supported dimension will retrieve in next release");
+		}
+		return new Dimension[] {getResolution()};
+	}
+
+	@Override
+	public boolean isOpen() {
+		return isOpen;
+	}
+	/**
+	 * start raspistill process to open devices 
+	 * step 1: check given arguments will introduce native windows. keep raspistill run in quietly
+	 * step 2: create thread fixed size pool(size=2), one for read, one for write
+	 * @see com.github.sarxos.webcam.WebcamDevice#open()
+	 */
+	@Override
+	public void open() {
+		if(isOpen()) {
+			LOGGER.warn("device already opened");
+			return ;
+		}
+		
+		service = Executors.newScheduledThreadPool(2, (Runnable r) -> {
+            Thread thread=new Thread(threadGroup(), r, "raspistill-device-"+threadId());
+            return thread;
+        });
+		
+		//no preview window
+		arguments.remove("preview");
+		arguments.remove("fullscreen");
+		arguments.remove("opacity");
+		arguments.remove("help");
+		arguments.remove("set");
+		arguments.put("output", "-");//must be this, then image will be in console!
+		
+		try {
+			process=launch();
+		} catch (IOException e) {
+			if(process!=null) {
+				try {
+					process.destroy();
+				} catch (Exception e1) {
+					LOGGER.warn("there maybe resource link, please report to issue list");
+					e1.printStackTrace();
+				}
+			}
+			e.printStackTrace();
+		}
+		
+		out = process.getOutputStream();
+        in = process.getInputStream();
+        err = process.getErrorStream();
+        
+        //this is one dummy work content that flash the process output stream
+        service.scheduleAtFixedRate(()->{
+        	try {
+				out.flush();
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+        }, 5, 5, TimeUnit.SECONDS);
+        
+        service.submit(()->{
+        	while (!Thread.currentThread().isInterrupted()) {
+        		//TODO read console to split images
+        	}
+        });
+        
+        isOpen=true;
+	}
+	
+	/**
+	 * @throws IOException  
+	 * 
+	 */
+	private Process launch() throws IOException {
+		StringBuilder command=new StringBuilder(12+arguments.size()*8);
+		command.append("raspistill ");
```

Comment:
raspistill can be extracted as a constant as it is called couple times

Reply:
ok